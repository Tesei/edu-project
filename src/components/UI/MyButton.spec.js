import { describe, it, expect, beforeAll, beforeEach, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import component from './MyButton.vue'

describe('Button component', () => {
    const buildWrapper = (hasAllData) => {
        const propsData = {
            hasAllData,
        }
        return mount(component, {
            propsData,
        })
    }
    it('mounts without error', () => {
        const wrapper = buildWrapper()
        expect(wrapper.exists).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('don have data - disabled button', () => {
        const wrapper = buildWrapper(false)
        expect(wrapper.find('button').attributes().disabled).toBe('')
    })
    it('if we have all data button - active', () => {
        const wrapper = buildWrapper(true)
        expect(wrapper.find('button').attributes().class).includes('_active')
    })
    it('button has attribute submit', () => {
        const wrapper = buildWrapper(true)
        expect(wrapper.find('button').attributes().type).includes('submit')
    })
})
