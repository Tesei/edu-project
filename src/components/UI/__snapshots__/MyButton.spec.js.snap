// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > mounts without error 1`] = `"<div data-v-8ee70a59="" class="button"><button data-v-8ee70a59="" type="submit" class="btn _active"></button></div>"`;
